anthos-bm-gcp-bash/install_admin_cluster.sh

```diff
@@ -44,286 +44,357 @@ fi
-printf "🔄 Adding IAM roles to the Service Account...\n"
-# [START anthos_bm_gcp_bash_admin_add_iam_role]
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/gkehub.connect" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/gkehub.admin" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/logging.logWriter" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/monitoring.metricWriter" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/monitoring.dashboardEditor" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/stackdriver.resourceMetadata.writer" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/opsconfigmonitoring.resourceMetadata.writer" \
-  --no-user-output-enabled
-# [END anthos_bm_gcp_bash_admin_add_iam_role]
-printf "✅ Successfully added the requires IAM roles to the Service Account.\n\n"
-
-# declare arrays for VM names and IPs
-printf "🔄 Setting up array variables for the VM names and IP addresses...\n"
-# [START anthos_bm_gcp_bash_admin_vms_array]
-MACHINE_TYPE=n1-standard-8
-VM_PREFIX=abm
-VM_WS=$VM_PREFIX-ws
-VM_ADMIN_CP=$VM_PREFIX-admin-cluster-cp
-VM_USER_CP=$VM_PREFIX-user-cluster-cp
-VM_USER_W1=$VM_PREFIX-user-cluster-w1
-VM_USER_W2=$VM_PREFIX-user-cluster-w2
-declare -a VMs=("$VM_WS" "$VM_ADMIN_CP" "$VM_USER_CP" "$VM_USER_W1" "$VM_USER_W2")
-declare -a IPs=()
-# [END anthos_bm_gcp_bash_admin_vms_array]
-printf "✅ Variables for the VM names and IP addresses setup.\n\n"
-
-# create GCE VMs
-printf "🔄 Creating GCE VMs...\n"
-# [START anthos_bm_gcp_bash_admin_create_vm]
-for vm in "${VMs[@]}"
+# --------------------------------------------------------------
+# This section checks what portions of the script the user wants
+# to execute. The script does two things. (1) Set up the GCE
+# infrastrcuture where the Anthos bare metal cluster will be
+# installed and (2) Install the Anthos bare metal cluster. Thus,
+# in this section we provide the user with the option to do run
+# everything or to only one of these two things.
+# --------------------------------------------------------------
+
+RUN_ALL="All-(Setup-and-Install)"
```

Comment:
A bit more info in the menu would be helpful.  Something like:

RUN_ALL="Set up the GCE infrastructure and install the admin cluster"
SETUP_ONLY="Set up the GCE infrastructure only"
INSTALL_ONLY="Install the admin cluster (assumes the GCE infrastructure has already been set up)"

Reply:
These variables are not printed on screen and the user will not be able to see them. Also these are used as matches for Switch-Case block; thus having spaces in between will not work.